packages/web3torrent/src/clients/web3t-channel-client.ts

```diff
@@ -0,0 +1,193 @@
+    const allocations = formatAllocations(aOutcomeAddress, bOutcomeAddress, aBal, bBal);
+    const appDefinition = '0x0'; // TODO SingleAssetPayments address
+
+    const channelResult = await this.channelClient.createChannel(
+      participants,
+      allocations,
+      appDefinition,
+      'appData unused'
+    );
+
+    return convertToChannelState(channelResult);
+  }
+
+  async getAddress() {
+    this.mySigningAddress = await this.channelClient.getAddress();
+    return this.mySigningAddress;
+  }
+
+  async getEthereumSelectedAddress() {
+    this.myEthereumSelectedAddress = await this.channelClient.getEthereumSelectedAddress();
+    return this.myEthereumSelectedAddress;
+  }
+
+  onMessageQueued(callback: (message: Message) => void) {
+    return this.channelClient.onMessageQueued(callback);
+  }
+
+  // Accepts an web3t-friendly callback, performs the necessary encoding, and subscribes to the channelClient with an appropriate, API-compliant callback
+  onChannelUpdated(web3tCallback: (channelState: ChannelState) => any) {
+    function callback(channelResult: ChannelResult): any {
+      web3tCallback(convertToChannelState(channelResult));
+    }
+    // These are two distinct events from the channel client
+    // but for our purposes we can treat them the same
+    // and rely on the channel status
+    const unsubChannelUpdated = this.channelClient.onChannelUpdated(callback);
+    const unsubChannelProposed = this.channelClient.onChannelProposed(callback);
+
+    return () => {
+      unsubChannelUpdated();
+      unsubChannelProposed();
+    };
+  }
+
+  async joinChannel(channelId: string) {
+    const channelResult = await this.channelClient.joinChannel(channelId);
+    return convertToChannelState(channelResult);
+  }
+
+  async closeChannel(channelId: string): Promise<ChannelState> {
+    const channelResult = await this.channelClient.closeChannel(channelId);
+    return convertToChannelState(channelResult);
+  }
+
+  async challengeChannel(channelId: string): Promise<ChannelState> {
+    const channelResult = await this.channelClient.challengeChannel(channelId);
+    return convertToChannelState(channelResult);
+  }
+
+  async updateChannel(
+    channelId: string,
+    aAddress: string,
+    bAddress: string,
+    aBal: string,
+    bBal: string,
+    aOutcomeAddress: string = aAddress,
+    bOutcomeAddress: string = bAddress
+  ) {
+    const allocations = formatAllocations(aOutcomeAddress, bOutcomeAddress, aBal, bBal);
+    const participants = formatParticipants(aAddress, bAddress, aOutcomeAddress, bOutcomeAddress);
+
+    // ignore return val for now and stub out response
+    const channelResult = await this.channelClient.updateChannel(
+      channelId,
+      participants,
+      allocations,
+      'appData unused'
+    );
+
+    return convertToChannelState(channelResult);
+  }
+
+  // seeder may use this method to accept payments
+  async counterSign(channelState: ChannelState) {
```

Comment:
Perhaps rename to `acceptPayment` which defines the "action" taken by the seeder more closely.